src/hooks/useGasLimit.ts

```diff
@@ -60,12 +64,23 @@ const useGasLimit = (
   const [gasLimit, gasLimitError, gasLimitLoading] = useAsync<BigNumber>(() => {
     if (!safeAddress || !walletAddress || !encodedSafeTx || !web3ReadOnly) return
 
-    return web3ReadOnly.estimateGas({
-      to: safeAddress,
-      from: walletAddress,
-      data: encodedSafeTx,
-      type: operationType,
-    })
+    return web3ReadOnly
+      .estimateGas({
+        to: safeAddress,
+        from: walletAddress,
+        data: encodedSafeTx,
+        type: operationType,
+      })
+      .then((gasLimit) => {
+        // Due to a bug in Nethermind estimation we need to increment 30% the gasLimit
+        // when the safeTxGas is defined and not 0.
+        const safeTxGas = safeTx?.data?.safeTxGas
+        if (safeTxGas && safeTxGas !== 0 && currentChainId === GNOSIS_CHAIN_ID) {
+          return gasLimit.mul(130).div(100)
+        }
```

Comment:
Since it's just for GC, I would move the chain id check to the beginning.

```suggestion
        if (currentChainId === chains.gno) {
          const gasLimitFactor = 1.3
          const safeTxGas = safeTx?.data?.safeTxGas
          if (safeTxGas && safeTxGas !== 0) {
            return gasLimit.mul(gasLimitFactor * 100).div(100)
          }
        }
```

Also, does `mul` accept only integers? Otherwise you could do `mul(1.3)`.

Reply:
Yes, mul only accepts integers, I already tested that

Reply:
The difference in computational performance would be infinitesimally small. The difference in readability, IMHO, is significant. A separate if would show clearly that that entire block of code is only a workaround for GC.